tests/constraints/models.py

```diff
@@ -28,6 +29,46 @@ class Meta:
         ]
 
 
+class GeneratedFieldStoredProduct(models.Model):
```

Comment:
Just a thought, see https://github.com/django/django/pull/18407 on some tests where the model is defined in the test cases with `db_persist` as a parameter (might be worth seeing if you like it for your tests as it might save defining two very similar models

Reply:
Thanks @sarahboyce, I tried this, but on saving an instance of a class defined within a test I receive the error below. I tried decorating the test method with `@isolate_apps("constraints")` but that didn't resolve the problem. Do you have any suggestions for how to work around this? If I can't resolve I'll just leave the tests as they were.



```
ERROR: test_validate_expression_generated_field_stored (constraints.tests.UniqueConstraintTests.test_validate_expression_generated_field_stored)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/melipone/PycharmProjects/django/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
    ^^^^^^^^^^^^^^^^^
  File "/home/melipone/PycharmProjects/django/.venv/lib/python3.12/site-packages/psycopg/cursor.py", line 97, in execute
    raise ex.with_traceback(None)
    ^^^^^^^^^^^^^^^^^
psycopg.errors.UndefinedTable: relation "constraints_testmodel" does not exist
LINE 1: INSERT INTO "constraints_testmodel" ("price", "discounted_pr...

```

Reply:
Leave it how it is :+1: I can have a look later but we may keep it how you have it anyway so don't worry too much

Reply:
Will, do thanks :-)